src/bun.js/bindings/URLDecomposition.cpp

```diff
@@ -35,7 +35,14 @@ String URLDecomposition::origin() const
 
     if (fullURL.protocolIsInHTTPFamily() or fullURL.protocolIsInFTPFamily() or fullURL.protocolIs("ws"_s) or fullURL.protocolIs("wss"_s))
         return fullURL.protocolHostAndPort();
-
+    if(fullURL.protocolIsBlob()) {
+        const String& path = fullURL.path().toString();
+        const URL subUrl { URL {}, path };
+        if(subUrl.isValid()) {
```

Comment:
hello, I'm not a maintainer or anything, but just noticed both your `if` (this one and the top one) are missing a space and that would probably fail Prettier linting. Also maybe wonder why you have 2 nested `if` when that could probably be just one combined, even though it might be a long condition

Reply:
i can bun fmt, but will fix ty, the seperation for the branches is because they concern different things.
In this case it does make sense to keep them separated